src/core/src/bootstrap/EnvLayer.py

```diff
@@ -200,6 +207,85 @@ def check_sudo_status(self, raise_if_not_sudo=True):
             if raise_if_not_sudo:
                 raise
 
+    def ensure_tty_not_required(self):
+        """ Checks current tty settings in /etc/sudoers and disables it within the current user context, if required. Sudo commands don't execute if tty is required. """
+        try:
+            tty_required = self.is_tty_required()
+            if tty_required:
+                self.disable_tty_for_current_user()
+        except Exception as error:
+            # todo: raise error?
+            print("Error occurred while ensuring tty is disabled. [Error={0}]".format(repr(error)))
+
+    def is_tty_required(self):
+        """ Checks if tty is set to required within the VM and will be applicable to the current user (either via a generic config or a user specific one) """
+        if self.is_tty_required_in_sudoers():
+            if self.is_tty_disabled_in_linux_patch_extension_sudoers():
+                return False
+            return True
+        return False
+
+    def is_tty_required_in_sudoers(self):
+        """ Reads the default tty setting from /etc/sudoers """
+        try:
+            tty_set_to_required = False
+            sudoers_default_configuration = self.file_system.read_with_retry(self.etc_sudoers_file_path)
+            settings = sudoers_default_configuration.strip().split('\n')
+
+            for setting in settings:
+                if self.require_tty_setting not in str(setting):
+                    continue
+
+                if re.match('.*!' + self.require_tty_setting, setting):
+                    defaults_set_for = re.search('(.*)!' + self.require_tty_setting, setting).group(1)
+                    if self.get_current_user() in defaults_set_for or 'Defaults' == defaults_set_for.strip():
+                        tty_set_to_required = False
+                else:
+                    defaults_set_for = re.search('(.*)' + self.require_tty_setting, setting).group(1)
+                    if self.get_current_user() in defaults_set_for or 'Defaults' == defaults_set_for.strip():
+                        tty_set_to_required = True
+
+            return tty_set_to_required
+
+        except Exception as error:
+            print("Error occurred while fetching data from [FilePath={0}] [Exception={1}]".format(str(self.etc_sudoers_file_path), repr(error)))
+            raise
```

Comment:
Checking basic tty belongs here but the rest of it (mitigation and checking for mitigation) doesn't logically fall here. There is a logical class of things that address issues in customer environments safely and as a separate unit. This belongs in such a class. However, due to how primitive this is in the execution flow, it may deserve an exception to sit directly inside bootstrapper (all of it) - since it may be considered a basic, self-contained portion of bootstrapping.

Reply:
The check_sudo_status function may also be moved into Bootstrapper going by that logical convention. It doesn't belong where it is currently looking at all of it now.

Reply:
Resolved